packages/docusaurus-theme-classic/src/theme/TOC/index.tsx

```diff
@@ -24,10 +25,16 @@ const TOC_HIGHLIGHT_PARAMS: TOCHighlightParams = {
 export function TOCHeadings({
   toc,
   isChild,
+  depth: headingLevel,
 }: TOCHeadingsProps): JSX.Element | null {
+  const {tableOfContents} = useThemeConfig();
+  const maxDepth = tableOfContents?.maxDepth ?? 3;
```

Comment:
Since you've already added a default value in Joi validation, there's no need to duplicate it here (Joi validation actually mutates the object)
Would it make sense to add a `minDepth` option as well? Would the user sometimes not want to have `h2` headings?

Reply:
> Since you've already added a default value in Joi validation, there's no need to duplicate it here (Joi validation actually mutates the object)

Weird because I get the following error when I don't have the null coalescing part:

```
TypeError: Cannot destructure property 'maxDepth' of 'tableOfContents' as it is undefined.
```

![image](https://user-images.githubusercontent.com/16010076/133697967-a6bd1c78-5a38-4efd-98de-f6955ad89e83.png)


`git diff` of the code change:
```diff
diff --git a/packages/docusaurus-theme-classic/src/theme/TOC/index.tsx b/packages/docusaurus-theme-classic/src/theme/TOC/index.tsx
index 83c3d2930..4dfe9038c 100644
--- a/packages/docusaurus-theme-classic/src/theme/TOC/index.tsx
+++ b/packages/docusaurus-theme-classic/src/theme/TOC/index.tsx
@@ -28,7 +28,7 @@ export function TOCHeadings({
   depth: headingLevel,
 }: TOCHeadingsProps): JSX.Element | null {
   const {tableOfContents} = useThemeConfig();
-  const maxDepth = tableOfContents?.maxDepth ?? 3;
+  const { maxDepth } = tableOfContents;
   if (!toc.length) {
     return null;
   }
```

Reply:
> Would it make sense to add a minDepth option as well? Would the user sometimes not want to have h2 headings?

Personally I think this feels like an anti-pattern since you generally want to utilize all markdown heading levels. I could see it for some niche cases like having a single `h2` on a page and not wanting to nest all of your `h3` inside, but feels funny to have it on site-wide in the theme config.

Happy to implement it if others feel differently.

Reply:
> Weird because I get the following error when I don't have the null coalescing part

Oh yeah, you need to provide a default value for the entire `tableOfContents` config object as well

Reply:
> but feels funny to have it on site-wide in the theme config.

I'm thinking about adding a per-doc option `minLevel` as well, and just for the sake of uniformity, I think a site-level config is worth it. But seems @slorber doesn't particularly like too many APIs, so let's see🤷‍♂️

Reply:
> Since you've already added a default value in Joi validation, there's no need to duplicate it here (Joi validation actually mutates the object)

Addressed this in 0785423

Reply:
For consistency I'd rather have min/max everywhere, even if min is less likely to be used and mostly for weird use-cases